analyzers/src/SonarAnalyzer.CSharp/Rules/DoNotCheckZeroSizeCollection.cs

```diff
@@ -126,18 +104,34 @@ private void AnalyzeSwitchStatement(SyntaxNodeAnalysisContext c)
             }
         }
 
-        private void CheckPatternCondition(SyntaxNodeAnalysisContext context, ExpressionSyntax expression, SyntaxNode pattern)
+        private void CheckPatternCondition(SyntaxNodeAnalysisContext context, SyntaxNode expression, SyntaxNode pattern)
         {
-            var relationalOrSubPattern = pattern.DescendantNodesAndSelf().FirstOrDefault(x => x.IsAnyKind(SyntaxKindEx.RelationalPattern, SyntaxKindEx.Subpattern));
-            if (RelationalPatternSyntaxWrapper.IsInstance(relationalOrSubPattern)
-                && ((RelationalPatternSyntaxWrapper)relationalOrSubPattern) is var relationalPattern
-                && IsOperatorOfInterest(relationalPattern.OperatorToken))
+            if (pattern.DescendantNodesAndSelf().FirstOrDefault(x => x.IsAnyKind(SyntaxKindEx.RelationalPattern, SyntaxKindEx.Subpattern)) is { } relationalOrSubPattern
+                && RelationalPatternSyntaxWrapper.IsInstance(relationalOrSubPattern)
+                && ((RelationalPatternSyntaxWrapper)relationalOrSubPattern) is { } relationalPattern
+                && ComparisonKind(relationalPattern.OperatorToken) is { } comparison
+                && comparison != Comparison.None
+                && Language.ExpressionNumericConverter.TryGetConstantIntValue(context.SemanticModel, relationalPattern.Expression, out var constant))
             {
-                CheckCondition(context, relationalPattern, expression, relationalPattern.Expression);
+                CheckExpression(
+                    context,
+                    relationalPattern.SyntaxNode,
+                    expression,
+                    constant,
+                    comparison);
             }
-        }
 
-        private static bool IsOperatorOfInterest(SyntaxToken syntaxToken) =>
-            syntaxToken.ValueText.Equals(">=") || syntaxToken.ValueText.Equals("<");
+            static ComparisonKind ComparisonKind(SyntaxToken syntaxToken) =>
+                syntaxToken.ValueText switch
```

Comment:
`==`, `!=`, `<=`, `>` and `_` cases are not covered. Can you add test cases, please?

Reply:
Sure.

Reply:
The `==` and `!=` are actually not valid. Do we need them in the implementation then?

Reply:
@pavel-mikula-sonarsource I'm not sure, Is there an other way to use these patterns, I'm not aware off?